components/admin/lessons/AdminLessonChallenges.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from 'react'
+import { useMutation } from '@apollo/react-hooks'
+import updateChallenge from '../../../graphql/queries/updateChallenge'
+import { FormCard } from '../../FormCard'
+import createNewChallenge from '../../../graphql/queries/createChallenge'
+import { Lesson, Challenge, Maybe } from '../../../graphql/index'
+import {
+  getPropertyArr,
+  checkForErrors,
+  checkForAllErrors,
+  makeGraphqlVariable
+} from '../../../helpers/admin/lessonHelpers'
+import { titleStyle } from './AdminLessonInfo'
+
+const challengeAttributes = {
+  title: '',
+  order: '',
+  description: '',
+  id: ''
+}
+
+type LessonChallengeProps = {
+  challenge: Maybe<Challenge>
+  alter: (options: any) => Promise<void>
+}
+
+type NewChallengeProps = {
+  setLessons: React.Dispatch<React.SetStateAction<Lesson[] | null>>
+  lessonId: number
+}
+type LessonChallengesProps = {
+  setLessons: React.Dispatch<React.SetStateAction<Lesson[] | null>>
+  challenges: Maybe<Challenge>[] | null | undefined
+  lessonId?: number
+}
+
+// Renders when someone clicks on `create new button` on the sidebar
+export const NewChallenge: React.FC<NewChallengeProps> = ({
+  setLessons,
+  lessonId
+}) => {
+  const [createChallenge, { loading, data }] = useMutation(createNewChallenge)
+  const [challengeProperties, setChallengeProperties] = useState(
+    getPropertyArr(challengeAttributes)
+  )
+  // when data is fully loaded after sending mutation request, update front-end lessons info
+  useEffect(() => {
+    !loading && data && setLessons(data.createChallenge)
+  }, [data])
+
+  // alter gets called when someone clicks button to create a lesson
+  const alter = async () => {
+    const newProperties = [...challengeProperties]
+    const errors = checkForAllErrors(newProperties)
+    if (errors) {
+      setChallengeProperties(newProperties)
+      return
+    }
+
+    try {
+      await createChallenge(
+        makeGraphqlVariable(challengeProperties, { lessonId })
+      )
+
+      window.location.reload()
```

Comment:
Why do we force a refresh after creating a challenge?

Reply:
There was a problem whenever I clicked the submit button to create a challenge. It would cause the page to not function correctly anymore. After trying a long to figure out without coming to a solution, song suggested for me to just reload the page for now since this `admin lessons` page will not be used very often.

Reply:
created an issue: https://github.com/garageScript/c0d3-app/issues/368